### ISSUE TITLE
Table of contents links don't work on mobile

### ISSUE THREAD

**Issue by tedmiston** [tedmiston/notes#7]
Perhaps the auto generated IDs are different or something.

**Comment by tedmiston** [comment]
This is a known issue with GitHub's markdown library (see github/markup#425).

For example, this markdown snippet (1), accessible with fragment (2), generates html (3).  But navigation to the fragment this way works only on desktop, not on mobile.

``` markdown
## 1. Meta
```

``` markdown
1. [Meta](#1-meta)
```

``` html
<h2>
    <a id="user-content-1-meta" class="anchor" href="#1-meta" aria-hidden="true"></a>
    1. Meta
</h2>
```

---

Anchors manually defined with the `name` attribute seem to have the same issue.  For example, my file [Remote.md](https://github.com/tedmiston/notes/blob/7edabc9bdd990e0d7c8b9236eee88ff3473e100a/books/Remote.md) uses (1) to link to (2) which generates html (3):

``` markdown
- [Introduction](#pre2)
```

``` markdown
## <a name="pre2"></a>Introduction
```

``` html
<h2>
    <a id="user-content-introduction" class="anchor" href="#introduction" aria-hidden="true"></a>
    <a name="user-content-pre2"></a>
    Introduction
</h2>
```

GitHub's name wrangling is breaking things.  You can choose either to: (1) comply with their JavaScript generated names and break elsewhere or (2) break on GitHub mobile.

Accessing the way above _should_ work because GitHub _should not_ change `name`s in post-processing, but they do.  It doesn't work because they change `name` on the anchor tags being scrolled to, but not on the anchors referencing them.  [why not?]

So in that sense, the behavior of the generated HTML on mobile complies with the HTML 5 spec for [navigating to a fragment identifier](http://stackoverflow.com/a/484781/149428).  And their JavaScript library _on desktop_ supports additional (non-compliant) behavior because they do name wrangling that isn't mentioned in their Markdown guides.

---

A temporary _but inelegant_ workaround is to [prefix fragments with `user-content-`](https://github.com/github/markup/issues/425#issuecomment-65798199), for example:

``` markdown
[foo](#foo)
```

becomes

``` markdown
[foo](#user-content-foo)
```

Of course this comes at the tradeoff identified above should you desire to render to html for use outside of GitHub.

**Comment by tedmiston** [comment]
Fixed by 3aa437e59e98cd8fbdb3020c0ce7553d5a431318
